File: ji/proj_2/app/src/main/java/com/example/proj_2/galleryAdapter.java
package com.example.proj_2;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;

import java.util.ArrayList;
import java.util.List;


public class galleryAdapter extends RecyclerView.Adapter<galleryAdapter.ImageViewHolder> {
    private ArrayList<MediaFileData> dataset;
    List<Integer> countImages;

    galleryAdapter(ArrayList<MediaFileData> list){
        this.dataset = list;
    }

    class ImageViewHolder extends RecyclerView.ViewHolder {
        public ImageView grdImg;
        public TextView gridName;
        public TextView gridSub;

        public ImageViewHolder(View view) {
            super(view);
            this.grdImg = view.findViewById(R.id.gridImg);
            this.gridName = view.findViewById(R.id.gridName);
            this.gridSub = view.findViewById(R.id.gridSub);
        }
    }

    @NonNull
    @Override
    public ImageViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View adapterLayout = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_folder_image, parent, false);
        return new ImageViewHolder(adapterLayout);
    }

    @Override
    public void onBindViewHolder(@NonNull ImageViewHolder holder, int position) {
        MediaFileData item = dataset.get(position);
        holder.gridName.setText(item.bucketId.toString());
        holder.gridSub.setText(countImages.get(position));

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //Toast.makeText(context, "Clicked", LENGTH_SHORT).show()
                //Intent intent = new Intent(context, OpenGalleryFolderActivity.class);
                //intent.putExtra("bucketId", item.bucketId);
                //intent.putExtra("bucketName", item.bucketName);
                //context.startActivity(intent);
            }
        });

        Glide.with(holder.itemView)
                .load(item.uri)
                .thumbnail(0.33f)
                .centerCrop()
                .into((ImageView) holder.itemView);
    }

    @Override
    public int getItemCount() {
        return dataset.size();
    }
}

File: sj/project1_final/app/src/main/java/com/example/project1_final/adapter/GalleryAdapter.kt
package com.example.project1_final.adapter

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import android.widget.Toast.LENGTH_SHORT
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.project1_final.OpenGalleryFolderActivity
import com.example.project1_final.R
import com.example.project1_final.model.MediaFileData

class GalleryAdapter(private val context: Context, private val dataset: List<MediaFileData>, private val countImages:List<Int>): RecyclerView.Adapter<GalleryAdapter.ImageViewHolder>() {
    class ImageViewHolder(private val view: View): RecyclerView.ViewHolder(view){
        val imageView: ImageView = view.findViewById(R.id.gridImg)
        val gridName: TextView = view.findViewById(R.id.gridName)
        val gridSub: TextView = view.findViewById(R.id.gridSub)

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ImageViewHolder {
        val adapterLayout = LayoutInflater.from(parent.context).inflate(R.layout.item_folder_image, parent, false)
        return ImageViewHolder(adapterLayout)
    }


    override fun onBindViewHolder(holder: ImageViewHolder, position: Int) {
        val item = dataset.getOrNull(position)
        holder.gridName.text = item!!.bucketName
        holder.gridSub.text = countImages[position].toString()

        holder.imageView.setOnClickListener {
            //Toast.makeText(context, "Clicked", LENGTH_SHORT).show()
            val intent:Intent = Intent(context, OpenGalleryFolderActivity::class.java)
            intent.apply{
                putExtra("bucketId", item.bucketId)
                putExtra("bucketName", item.bucketName)
            }
            context.startActivity(intent)
        }

        Glide.with(holder.imageView)
            .load(item!!.uri)
            .thumbnail(0.33f)
            .centerCrop()
            .into(holder.imageView)
    }

    override fun getItemCount(): Int {
        return dataset.size
    }
}